### PR TITLE
verify and update protobufs in make verify/update

### PR DESCRIPTION
**Pull request by bparees**

### CODE REVIEW

**Comment by bparees** [comment]
@smarterclayton @stevekuznetsov is there a reason we weren't already doing this? ptal.

**Comment by bparees** [comment]
[test]

**Comment by bparees** [comment]
looks like at least one reason is our AMIs don't have the right protoc on them......but that should be fixable.

**Comment by smarterclayton** [comment]
Because we do it in test-end-to-end because installing protoc is is a bitch
and we use the docker image.

So we should already be doing this in all scenarios.

**Comment by bparees** [comment]
well i spent several hours last night debugging an issue that was caused by not regenning the protobufs (but i had run make update), so i'd be in favor of us finding a way to include this, or at least warning people that make update doesn't actually cover everything they need to do when editing apis.

**Comment by smarterclayton** [comment]
We can make it conditional and warn if you don't have proto.  We are trying
move all tools out of the VMs and into the release image though so this is
a temporary thing.  We can't keep installing crap onto our vms

On Sep 21, 2016, at 11:35 AM, Ben Parees <notifications@github.com> wrote:

well i spent several hours last night debugging an issue that was caused by
not regenning the protobufs (but i had run make update), so i'd be in favor
of us finding a way to include this, or at least warning people that make
update doesn't actually cover everything they need to do when editing apis.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/openshift/origin/pull/11036#issuecomment-248650537>,
or mute
the thread

**Comment by bparees** [comment]
I'll work on that.

Ben Parees | OpenShift

**Comment by bparees** [comment]
@smarterclayton @stevekuznetsov updated, ptal.
[test]

File: hack/update-generated-protobuf.sh [hunk 1]
```diff
@@ -7,6 +7,9 @@ if [[ -z "$(which protoc)" || "$(protoc --version)" != "libprotoc 3.0."* ]]; the
   echo
   echo "  https://github.com/google/protobuf/releases"
   echo
+  if [[ ${PROTO_OPTIONAL:-} == "1" ]]; then
```
Comment (stevekuznetsov) [inline hack/update-generated-protobuf.sh]:
```sh
if [[ "${PROTO_OPTIONAL:-}" == "1" ]]; then
      ^                   ^
```

Comment (bparees) [inline hack/update-generated-protobuf.sh]: done.

Comment (stevekuznetsov) [inline hack/update-generated-protobuf.sh]: Do these review comments not go away once they're addressed? Weird.

**Comment by stevekuznetsov** [comment]
Looks reasonable to me.

**Comment by smarterclayton** [comment]
Lgtm [merge]

**Pull request merged**